### PR TITLE
Adding failover marker replication task in thrift and persistence struct

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Adding failover marker thrift struct and persistence struct
IDL PR: https://github.com/uber/cadence-idl/pull/31

**Why?**
To support graceful failover

**How did you test it?**
Tests will be added when there is business logic

**Potential risks**
N/A

### CODE REVIEW

File: service/history/historyEngine.go [hunk 1]
```diff
@@ -317,7 +317,7 @@ func (e *historyEngineImpl) Start() {
 	clusterMetadata := e.shard.GetClusterMetadata()
 	if e.replicatorProcessor != nil &&
 		(clusterMetadata.GetReplicationConsumerConfig().Type != sconfig.ReplicationConsumerTypeRPC ||
-		e.config.EnableKafkaReplication()) {
+			e.config.EnableKafkaReplication()) {
```
Comment (yycptt) [inline service/history/historyEngine.go:320]: Revert the change?

File: common/persistence/dataInterfaces.go [hunk 2]
```diff
@@ -605,6 +606,15 @@ type (
 		ScheduledID         int64
 	}
 
+	// FailoverMarkerTask is the marker for graceful failover
+	FailoverMarkerTask struct {
+		TaskID              int64
+		VisibilityTimestamp time.Time
+		Version             int64
+		SourceCluster       string
```
Comment (yycptt) [inline common/persistence/dataInterfaces.go:614]: Does the underlying DB schema has a field for the `sourceCluster`?

Comment (yux0) [reply common/persistence/dataInterfaces.go:614]: No. We don't have. Currently, I just re-use a string field.

File: service/history/historyEngine.go (hunk 1 above)
Comment (yux0) [reply service/history/historyEngine.go:320]: It is done by the formater

File: common/persistence/dataInterfaces.go (hunk 2 above)
Comment (yycptt) [reply common/persistence/dataInterfaces.go:614]: I see. Probably add some comments then?

Comment (yux0) [reply common/persistence/dataInterfaces.go:614]: I will update the schema in the following PR.
